### PR TITLE
[feat]: [CDS-41934] Git Triggers

### PR DESCRIPTION
**Pull request by vivekkarnatiharness**
## Describe your changes
Github Packages triggers backend changes

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37315)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 12 Issues 
* Code issue 12 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37315)

File: 125-cd-nextgen/src/main/java/io/harness/cdng/artifact/mappers/ArtifactConfigToDelegateReqMapper.java [hunk 1]
```diff
@@ -91,17 +91,8 @@ public S3ArtifactDelegateRequest getAmazonS3DelegateRequest(AmazonS3ArtifactConf
...
-    if (StringUtils.isBlank(versionRegex)) {
-      versionRegex = "";
-    }
-
-    String version = artifactConfig.getVersion().getValue();
-
-    if (StringUtils.isBlank(version)) {
-      version = "";
-    }
+    String versionRegex = artifactConfig.getVersionRegex() != null ? artifactConfig.getVersionRegex().getValue() : "";
```
Comment (puthrayaharness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/artifact/mappers/ArtifactConfigToDelegateReqMapper.java:94]: We need to do a blank check here.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/artifact/mappers/ArtifactConfigToDelegateReqMapper.java [hunk 2]
```diff
@@ -91,17 +91,8 @@ public S3ArtifactDelegateRequest getAmazonS3DelegateRequest(AmazonS3ArtifactConf
...
+    String version = artifactConfig.getVersion() != null ? artifactConfig.getVersion().getValue() : "";
```
Comment (puthrayaharness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/artifact/mappers/ArtifactConfigToDelegateReqMapper.java:95]: Please do a blank check

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Review by puthrayaharness — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 66 of 77 files at r4, 20 of 20 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37315)* status: :shipit: complete! ✓

**Review by archit-harness — APPROVED** [review]
> Reviewed 2 of 77 files at r4, 6 of 20 files at r5.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37315)* status: :shipit: complete! ✓

**Pull request merged**